services/cd-service/pkg/repository/transformer_db_test.go

```diff
@@ -3190,3 +3190,156 @@ func TestTransaction(t *testing.T) {
 		})
 	}
 }
+
+func TestTimestampConsistency(t *testing.T) {
+	tcs := []struct {
+		Name                 string
+		ReleaseVersionsLimit uint
+		Transformers         []Transformer
+		ExpectedVersion      uint
+		TargetEnv            string
+		TargetApp            string
+	}{
+		{
+			Name:            "Release train",
+			ExpectedVersion: 2,
+			TargetEnv:       envProduction,
+			TargetApp:       "test",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: envProduction,
+					Config: config.EnvironmentConfig{
+						Upstream: &config.EnvironmentConfigUpstream{
+							Environment: envAcceptance, // train drives from acceptance to production
+						},
+					},
+				},
+				&CreateEnvironment{
+					Environment: envAcceptance,
+					Config: config.EnvironmentConfig{
+						Upstream: &config.EnvironmentConfigUpstream{
+							Environment: envAcceptance,
+							Latest:      true,
+						},
+					},
+				},
+				&CreateApplicationVersion{
+					Application: "test",
+					Manifests: map[string]string{
+						envProduction: "productionmanifest",
+						envAcceptance: "acceptancenmanifest",
+					},
+					WriteCommitData: true,
+					Version:         1,
+				},
+				&DeployApplicationVersion{
+					Environment: envProduction,
+					Application: "test",
+					Version:     1,
+				},
+				&CreateApplicationVersion{
+					Application: "test",
+					Manifests: map[string]string{
+						envProduction: "productionmanifest",
+						envAcceptance: "acceptancenmanifest",
+					},
+					WriteCommitData: true,
+					Version:         2,
+				},
+				&DeployApplicationVersion{
+					Environment: envAcceptance,
+					Application: "test",
+					Version:     1,
+				},
+				&DeployApplicationVersion{
+					Environment: envAcceptance,
+					Application: "test",
```

Comment:
There are several parts that we used "test" for application name in this testcase. Maybe we can define a constant called applicationName at the top.